sources/core.ts

```diff
@@ -155,22 +185,40 @@ export function runMachineInternal(machine: StateMachine, input: Array<string>,
     positionals: [],
     remainder: null,
     selectedIndex: null,
+    completion: null,
   }}];
 
   debugMachine(machine, {prefix: `  `});
 
+  const isCompletionMode = typeof completionCursorPosition !== `undefined`;
+
+  let cursorPosition = -(START_OF_INPUT.length + 1);
```

Comment:
It's because all tokens are expected to be separated by a single space that's also taken into account when computing the cursor position. It's [here](https://github.com/paul-soporan/clipanion/blob/9b09ad3a4741935187ce90b94cc69b082a01a14e/sources/core.ts#L275) where we do a `cursorPosition += segment.length + 1` which also takes into account the space separator. I'll add a comment about this.